docs/specification/revisions/draft/auth/oauth2.md

```diff
@@ -0,0 +1,243 @@
+
+## Protocol Flow
+We follow the flows defined by [RFC 6749](https://datatracker.ietf.org/doc/html/rfc6749).
+
+```mermaid
+sequenceDiagram
+    participant Client as MCP Client
+    participant User as Resource Owner (End User)
+    participant AuthServer as Authorization<br> MCP Server
+    participant Resource as Resource<br>MCP Server
+
+    Note right of Client: Obtaining an access token:
+    Client->>User: Authorization Request (auth/oauth2/authorize)
+    User->>Client: Authorization Grant
+    Client->>AuthServer: Authorization Grant (auth/oauth2/token)
+    AuthServer->>Client: Access Token
+
+    Note right of Client: Using an access token:
+    Client->>Resource: Access prompts/resources/tools with Access Token
+    Resource->>Client: Receive protected response
+    
+
+    Note right of Client: Refreshing an access token:
+    Client->>AuthServer: Refresh Token Request (auth/oauth2/token)
+    AuthServer->>Client: New Access Token
+
+    Note right of Client: Revoking tokens:
+    Client->>AuthServer: Revoke Token Request (auth/oauth2/revoke)
+```
+Note that the authorization server may be the same server as the resource server or a separate entity. A single authorization server may issue access tokens accepted by multiple resource servers.
+
+## Capabilities
+
+
+Servers supporting OAuth 2.0 **MUST** include their capabilities:
+
+```json
+{
+  "capabilities": {
+    "auth": {
+      "oauth2": {
+        "authorize": {},
+        "token": {},
+        "revoke": {}
+      }
+    }
+  }
+}
+```
+
+## Flows
+### Initialization
+During initialization, if the server supports the `oauth2` capability, the client **SHOULD** include an access token in all subsequent requests. If the client does not have an access token, the client **SHOULD** obtain one.
+
+### Obtaining an Access Token
+
+#### Authorization Grant Flow
+To obtain an access token, clients **MUST** send an `auth/oauth2/authorize` request.
+
+**Request:**
+```typescript
+{
+  "jsonrpc": "2.0",
+  "id": 1,
+  "method": "auth/oauth2/authorize",
+  "params": {
+    "response_type": "code", // REQUIRED
+    "client_id": "client_id", // REQUIRED
+    "redirect_uri": "redirect_uri", // OPTIONAL
+    "scope": "scope", // OPTIONAL
+    "state": "state", // RECOMMENDED
+  }
+}
+```
+**Response:**
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 1,
+  "result": {
+    "authorization_url": "<scheme>://<domain>/oauth2"
+  }
+}
+```
+- The client **MUST** redirect the user to the authorization URL.
+- The authorization URL **SHOULD** prompt the user to configure any credentials as needed. Once complete, the server provides an authorization grant in the form of a code.
+- The client **MUST** implement a way to receive the authorization grant. This can be through a callback such as the redirect URI, providing some interface for the user to provide it, etc.
+
```

Comment:
With the above suggested change to redirect directly to /authorize, it is still the client's responsibility to receive the authorization grant (ie, the authorization code).  Now, however, there's no need for the intermediary JSON-RPC request.